lib/worker.js

```diff
@@ -0,0 +1,61 @@
+var debug       = require('debug')('github:worker');
+var tcrc        = require('./taskclusterrc');
+var github      = require('./github');
+var request     = require('request-promise')
+var taskcluster = require('taskcluster-client')
+var slugid      = require('slugid');
+
+var worker = module.exports = {};
+
+/**
+ * If a .taskclusterrc exists, attempt to turn it into a taskcluster
+ * graph config, and submit it to the scheduler.
+ **/
+worker.webHookHandler = async function(message, context) {
+  debug('handling pull request: ', message);
+  let msgPayload = message.payload;
+  // Try to fetch a taskclusterrc file for every request
+  let taskclusterrc = await request(tcrc.buildConfigUrl(msgPayload));
+  if (taskclusterrc) {
+    try {
+      let scheduler = new taskcluster.Scheduler(context.cfg.get('taskcluster'));
```

Comment:
Any value of adding a scheduler client to the context instead of instantiating a new one on each pulse message?  I ask because I'm genuinely curious for the things I do.